src/overlays/actors/ovl_En_Zod/z_en_zod.c

```diff
@@ -28,62 +60,592 @@ const ActorInit En_Zod_InitVars = {
+                    case 0x121C:
+                    case 0x1220:
+                    case 0x1221:
+                    case 0x1227:
+                        func_80151938(play, (play->msgCtx.currentTextId + 1) & 0xFFFF);
+                        break;
+
+                    case 0x1219:
+                        func_80151938(play, (play->msgCtx.currentTextId + 1) & 0xFFFF);
+                        this->nextAnimIndex = ENZOD_ANIM_ARMS_FOLDED;
+                        break;
+
+                    case 0x121D:
+                        func_80151938(play, (play->msgCtx.currentTextId + 1) & 0xFFFF);
+                        this->nextAnimIndex = ENZOD_ANIM_READY_TO_PLAY;
+                        break;
+
+                    case 0x1223:
+                        func_80151938(play, (play->msgCtx.currentTextId + 1) & 0xFFFF);
+                        this->nextAnimIndex = ENZOD_ANIM_PLAYING_VIVACE;
+                        break;
+
+                    case 0x121E:
+                    case 0x1226:
+                        func_80151938(play, 0x121F);
+                        this->nextAnimIndex = ENZOD_ANIM_READY_TO_PLAY;
+                        break;
+
+                    default:
+                        func_801477B4(play);
+                        this->actionFunc = EnZod_PlayDrumsSequence;
+                        EnZod_ChangeAnimation(this, ENZOD_ANIM_PLAYING_LENTO, ANIMMODE_ONCE);
+                        break;
+                }
+            }
+    }
+}
+
+void EnZod_PlayDrumsSequence(EnZod* this, PlayState* play) {
+    s32 pad;
+    Vec3f seqPos;
+
+    EnZod_UpdateAnimations(this);
+
+    if (Actor_ProcessTalkRequest(&this->actor, &play->state)) {
+        EnZod_RoomConversation(this, play);
+        this->actionFunc = func_80BAF7CC;
+    } else if (EnZod_PlayerIsFacingTijo(this, play)) {
+        func_800B8614(&this->actor, play, 210.0f);
+    }
+
+    seqPos.x = this->actor.projectedPos.x;
+    seqPos.y = this->actor.projectedPos.y;
+    seqPos.z = this->actor.projectedPos.z;
+
+    func_801A1FB4(3, &seqPos, NA_BGM_DRUMS_PLAY, 700.0f);
+}
+
+void func_80BAFA44(EnZod* this, PlayState* play) {
+    u16 textId;
+
+    if (gSaveContext.save.playerForm == PLAYER_FORM_ZORA) {
+        if (gSaveContext.save.weekEventReg[0x4F] & 1) {
+            textId = 0x1253; // And with that, we're all ready...
+        } else {
+            textId = 0x1251; // What have you been doing? You're late, Mikau!
+            if (gSaveContext.save.weekEventReg[0x4E] & 0x20) {
+                textId = 0x1252; // Hurry up and go see Lulu.
+            } else {
+                gSaveContext.save.weekEventReg[0x4E] |= 0x20;
+            }
+        }
+    } else {
+        textId = 0x1250; // If you want an autograph...
+    }
+
+    EnZod_ChangeAnimation(this, ENZOD_ANIM_PLAYING_VIVACE, ANIMMODE_ONCE);
+    Message_StartTextbox(play, textId, &this->actor);
+}
+
+void func_80BAFADC(EnZod* this, PlayState* play) {
+    u8 msgState;
```

Comment:
```suggestion
    u8 talkState;
```
We've started to standardize these